src/lib/parquet/src/Flow/Parquet/Reader.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types=1);
+
+namespace Flow\Parquet;
+
+use Flow\Parquet\Exception\InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+final class Reader
+{
+    public function __construct(
+        private readonly ByteOrder $byteOrder = ByteOrder::LITTLE_ENDIAN,
+        private Options $options = new Options(),
+        private readonly LoggerInterface $logger = new NullLogger()
+    ) {
+    }
+
+    public function read(string $path) : ParquetFile
+    {
+        if (!\file_exists($path)) {
+            throw new InvalidArgumentException("File {$path} does not exist");
+        }
+
+        $stream = \fopen($path, 'rb');
```

Comment:
I don't see any `\fclose` anywhere, aren't we leaving open file streams? Shouldn't `ParquetFile` implement `__destruct()` method with such `\fclose`?

Reply:
Destructor added 👍